src/redshift/IOP/FRI/coset_combining_fri/coset_combiner.rs

```diff
@@ -0,0 +1,95 @@
+/// CosetCombiner is a helper structure designed to solve the following problem: 
+/// assume we want to get the element at index [i] located in FRI oracle 
+/// however the values stored in FRI oracles are permuted, so that elements of the same coset are adjacent 
+/// So, we need to somehow get all elements of rhe same coset using natural index [i], i.e. 
+/// make a conversion from "natural" element index to "tree" coset index
+/// note, that this depends only on the number of elements in each coset i.e. "collapsing factor"
```

Comment:
Please, change to `//!` module docs.
https://doc.rust-lang.org/1.9.0/book/documentation.html#documenting-modules